File: frontend/src/models/calculator/CalculatorRequestModel.ts
import type { Sex } from "./enums/Sex.ts";
import type { ActivityLevel } from "./enums/ActivityLevel.ts";
import type { IsPregnant } from "./enums/IsPregnant.ts";
import type { IsBreastfeeding } from "./enums/IsBreastfeeding.ts";
import type { WeightGoal } from "./enums/WeightGoal.ts";

export type CalculatorRequestModel = {
  sex: Sex;
  age: number | null;
  weight: number | null;
  height: number | null;
  activityLevel: ActivityLevel;
  dailySport: boolean;
  isPregnant: IsPregnant;
  isBreastfeeding: IsBreastfeeding;
  weightGoal: WeightGoal;
  carbGoal: number;
  proteinGoalFactor: number;
};

File: backend/src/main/java/org/example/backend/calculator/model/calculator/CalculatorRequestModel.java
package org.example.backend.calculator.model.calculator;

import org.example.backend.calculator.model.enums.*;

public record CalculatorRequestModel(
        SexEnum sex,
        int age,
        double weight,
        int height,
        ActivityLevelEnum activityLevel,
        boolean dailySport,
        IsPregnantEnum isPregnant,
        IsBreastfeedingEnum isBreastfeeding,
        WeightGoalEnum weightGoal,
        int carbGoal,
        double proteinGoalFactor
) {
}
